climada/entity/impact_funcs/trop_cyclone.py

```diff
@@ -311,118 +436,70 @@ def get_countries_per_region(region=None):
-                "UMI", "ESH", "ZMB", "ALA",
-            ],
-        }
-        # fmt: on
-        impf_id = {
-            "NA1": 1,
-            "NA2": 2,
-            "NI": 3,
-            "OC": 4,
-            "SI": 5,
-            "WP1": 6,
-            "WP2": 7,
-            "WP3": 8,
-            "WP4": 9,
-            "ROW": 10,
-        }
-        region_name = dict()
-        region_name["NA1"] = "Caribbean and Mexico"
-        region_name["NA2"] = "USA and Canada"
-        region_name["NI"] = "North Indian"
-        region_name["OC"] = "Oceania"
-        region_name["SI"] = "South Indian"
-        region_name["WP1"] = "South East Asia"
-        region_name["WP2"] = "Philippines"
-        region_name["WP3"] = "China Mainland"
-        region_name["WP4"] = "North West Pacific"
 
         if region == "all":
-            return region_name, impf_id, iso3n, iso3a
+            return (
+                CountryCode.REGION_NAME.value,
+                CountryCode.IMPF_ID.value,
+                CountryCode.ISO3N.value,
+                CountryCode.ISO3A.value,
+            )
+
+        return (
+            CountryCode.REGION_NAME.value[region],
+            CountryCode.IMPF_ID.value[region],
+            CountryCode.ISO3N.value[region],
+            CountryCode.ISO3A.value[region],
+        )
+
+    @staticmethod
+    def get_regions_per_countries(
+        countries: list = None, code_type: str = "ISO3A"
+    ) -> tuple:
+        """Return the impact function id and the region corresponding to a list of countries,
+        or a single country.
+
+        Parameters:
+        -----------
+        countries : list
+            List containing the ISO code of the country, which should be either
+            a string if the code is iso3a or an integer if ISO3N. For example, for Switzerland:
+            the ISO3A code is "CHE" and the ISO3N is 756.
+        code_type : str
+            Either "ISO3A" or "ISO3N".
+
+        Returns:
+        --------
+        impf_ids : list
+            List of impact function ids matching the countries.
+        regions_ids : list
+            List of the regions that match the countries.
+        regions_names : list
+            List of the regions that match the countries.
+        """
 
-        return region_name[region], impf_id[region], iso3n[region], iso3a[region]
+        if code_type not in {"ISO3A", "ISO3N"}:
+            raise ValueError("code_type must be either 'iso3a' or 'iso3n'")
+        elif not all(isinstance(country, type(countries[0])) for country in countries):
+            raise ValueError("All elements in the list must be of the same type.")
+        elif code_type == "ISO3A" and isinstance((countries[0]), int):
+            raise ValueError("ISO3A code type cannot have integer values.")
+        elif code_type == "ISO3N" and isinstance((countries[0]), str):
+            raise ValueError("ISO3N code type cannot have string values.")
```

Comment:
PLease make clear error messages. Here it is neither consistent (e.e. iso3a and ISO3A), nor standard.